test/v200/utils/common/project_test_utils.go

```diff
@@ -0,0 +1,244 @@
+func getRemotes() map[string]string {
+	remotes := make(map[string]string)
+	numRemotes := GetRandomNumber(1, 5)
+	for i := 0; i < numRemotes; i++ {
+		key := GetRandomUniqueString(GetRandomNumber(6, 12), false)
+		remotes[key] = GetRandomUniqueString(GetRandomNumber(6, 12), false)
+		LogInfoMessage(fmt.Sprintf("Set remote key= %s, value= %s", key, remotes[key]))
+	}
+	return remotes
+}
+
+// AddProject adds a project of the specified type, with random attributes, to the devfile schema
+func (testDevfile *TestDevfile) AddProject(projectType schema.ProjectSourceType) string {
+	project := testDevfile.createProject(projectType)
+	testDevfile.SetProjectValues(&project)
+	return project.Name
+}
+
+// AddStarterProject adds a starter project of the specified type, with random attributes, to the devfile schema
+func (testDevfile *TestDevfile) AddStarterProject(projectType schema.ProjectSourceType) string {
+	starterProject := testDevfile.createStarterProject(projectType)
+	testDevfile.SetStarterProjectValues(&starterProject)
+	return starterProject.Name
+}
+
+// createProject creates a project of a specified type with only required attributes set
+func (testDevfile *TestDevfile) createProject(projectType schema.ProjectSourceType) schema.Project {
+	project := schema.Project{}
+	project.Name = GetRandomUniqueString(GetRandomNumber(8, 63), true)
+	LogInfoMessage(fmt.Sprintf("Create Project Name: %s", project.Name))
+
+	if projectType == schema.GitProjectSourceType {
+		project.Git = createGitProject()
+	} else if projectType == schema.GitHubProjectSourceType {
+		project.Github = createGithubProject()
+	} else if projectType == schema.ZipProjectSourceType {
+		project.Zip = createZipProject()
+	}
+	testDevfile.projectAdded(project)
+	return project
+}
+
+// generateStarterProject creates a starter project of a specified type with only required attributes set
+func (testDevfile *TestDevfile) createStarterProject(projectType schema.ProjectSourceType) schema.StarterProject {
+	starterProject := schema.StarterProject{}
+	starterProject.Name = GetRandomUniqueString(GetRandomNumber(8, 63), true)
+	LogInfoMessage(fmt.Sprintf("Create StarterProject Name: %s", starterProject.Name))
+
+	if projectType == schema.GitProjectSourceType {
+		starterProject.Git = createGitProject()
+	} else if projectType == schema.GitHubProjectSourceType {
+		starterProject.Github = createGithubProject()
+	} else if projectType == schema.ZipProjectSourceType {
+		starterProject.Zip = createZipProject()
+	}
+	testDevfile.starterProjectAdded(starterProject)
+	return starterProject
+
+}
+
+// createGitProject creates a git project structure with mandatory attributes set
+func createGitProject() *schema.GitProjectSource {
+	project := schema.GitProjectSource{}
+	project.Remotes = getRemotes()
+	return &project
+}
+
+// createGithubProject creates a github project structure with mandatory attributes set
+func createGithubProject() *schema.GithubProjectSource {
```

Comment:
Github projects are now deprecated (see https://github.com/devfile/api/issues/340), so I don't think we need to test it.

Reply:
OK, I will remove it

Reply:
Removed in latest commit